docs/localization.md

```diff
@@ -0,0 +1,144 @@
+
+Name your translation files based on the [BCP 47 specification](https://tools.ietf.org/html/bcp47) of language tags and region codes. For an overview of how these language tags are constructed, see [Language tags in HTML and XML](http://www.w3.org/International/articles/language-tags/).
+
+You can find a list of code subtags in the [IANA Language Subtag Registry](http://www.iana.org/assignments/language-subtag-registry). These subtags are primitives that you can combine to create file name prefixes for individual regions. Here are some examples:
+
+| Localization file name | Corresponding regional language variant | Subtags used |
+| ----------- | ----------- | ----------- |
+| `en.json` | English (default file)| en (English) |
+| `en-US.json` | American English | en (English) + US (United States) |
+| `en-AU.json` | Australian English | en (English) + AU (Australia) |
+| `fr.json` | French | en (French) |
+| `fr-CA.json` | Canadian French | fr (French) + CA (Canada) |
+
+### Translation keys
+
+The JSON files should contain key-value pairs for each locale. You can define translations based on a pre-defined keys used to translate the Catalyst storefront's [basic ecommerce functionality](https://www.catalyst.dev/docs#ecommerce-functionality). The translated values you specify will display to shoppers as static string translations.
+
+Use the existing `en.json` file as a template for the schema.
+You can only provide translated values for the translation keys specified in the template. 
+
+For example, the `en.json` file contains translation keys for 
+
+```json
+"Home": {
+  "Carousel": {
+    "bestSellingProducts": "Best Selling Products",
+    "featuredProducts": "Featured Products",
+    "newestProducts": "Newest products"
+  }
+}
+``` 
+
+In your newly-created JSON file, add a translation of the value to the new locale
+
+```json
+"Home": {
+  "Carousel": {
+    "bestSellingProducts": "Produits les plus vendus",
+    "featuredProducts": "Produits populaires",
+    "newestProducts": "Produits les plus recentsProduits"
+  }
+}
+``` 
+
+Read more about i18n messages in [next-intl docs](https://next-intl-docs.vercel.app/docs/usage/messages).
+
+## i18n configuration
+
+After you created a language file, add its name to the `i18n.ts` config file.
+
+For example, if you created a `fr.json` file, include the `fr` when you define the locales:   
+
+```ts
+const locales = ['en', 'fr'] as const;
+```
+
+## Using keys in React components
+
+The following example shows how messages can be used in **server** component:
+
+```tsx
+import { NextIntlClientProvider } from 'next-intl';
+import { getMessages, getTranslations, unstable_setRequestLocale } from 'next-intl/server';
+// ...
+import { LocaleType } from '~/i18n';
+
+interface Props {
+  params: {
+    locale: LocaleType;
+  };
+}
+
+export default async function Home({ params: { locale } }: Props) {
+  unstable_setRequestLocale(locale);
```

Comment:
@BC-krasnoshapka do we need to pay attention on using `unstable_setRequestLocale` and add a link to `next-intl` section regarding this method?